oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreBranch.java

```diff
@@ -122,9 +125,9 @@ public NodeState merge(@NotNull CommitHook hook, @NotNull CommitInfo info)
                 throw e;
             }
         }
-        // retry with exclusive lock, blocking other
+        // retry with exclusive lock (if avoidMergeLock is not enabled), blocking other
         // concurrent writes
-        return merge0(hook, info, true);
+        return merge0(hook, info, !avoidMergeLock);
```

Comment:
in case of `!avoidMergeLock` do you even want to do another retry? `merge0` already does a number of retries (depending on value of `maximumBackoff`) - I'd argue that if that fails and `avoidMergeLock` is true it should just rethrow the `e` above

Reply:
Done